source/extensions/common/wasm/context.cc

```diff
@@ -1110,6 +1110,12 @@ WasmResult Context::log(uint32_t level, absl::string_view message) {
   NOT_REACHED_GCOVR_EXCL_LINE;
 }
 
+uint32_t Context::getLogLevel() {
+  // Like the "log" call above, assume that spdlog level as an int
+  // matches the enum in the SDK
+  return static_cast<uint32_t>(ENVOY_LOGGER().level());
+}
```

Comment:
It might be a bit too late to ask this question, but how do you intend to use this?

Log level can change at runtime (at least in Envoy), so either you have to actively check the log level before each log attempt (expensive) or you're facing the possibility of not producing the log entries.

Wouldn't a `proxy_on_log_level_change` callback be more useful?

Reply:
I agree, but this is a start.